root/etc/cont-init.d/50-certbot

```diff
@@ -214,12 +217,11 @@ if [ ! "$URL" = "$ORIGURL" ] || [ ! "$SUBDOMAINS" = "$ORIGSUBDOMAINS" ] || [ ! "
     if [[ -f /config/etc/letsencrypt/live/"$ORIGDOMAIN"/fullchain.pem ]]; then
         certbot revoke --non-interactive --cert-path /config/etc/letsencrypt/live/"$ORIGDOMAIN"/fullchain.pem --server $REV_ACMESERVER
     fi
-    rm -rf /config/etc/letsencrypt
-    mkdir -p /config/etc/letsencrypt
+    rm -rf /config/etc/letsencrypt/{live,renewal}
```

Comment:
After reviewing what actually happens when you run `certbot delete`, these two folders are the only things touched. Normally `certbot delete` is used to remove one cert at a time, and this assumes you have the required `.conf` file in the `renewal` folder, and the `fullchain.pem` and other files in the `live` folder. In case these things get messed up, it's safer to remove the whole `live` and `renewal` folder.

certbot doesn't recommend this, but in the two scenarios where this script decides to remove these folders it makes sense, and will be more consistent with fixing user issues compared to using `certbot delete` in the recommended way.

p.s. I did write up the code changes to iterate through the output of `certbot certificates` and then use `certbot delete` on each one, but if the user's files and folders are messed up at all `certbot delete` won't handle it nicely, and the next run of `certbot renew` could attempt to renew a certificate with issues.

Reply:
Oh right, the reason for removing these subfolders and not the parent folder:
I want to keep the `/config/etc/letsencrypt/renewal-hooks` folder, which should allow for user modifications. I reviewed the other folders in `/config/etc/letsencrypt` and they don't seem like they will cause any issues being left behind.